lib/improver/metadata/check_datatypes.py

```diff
@@ -34,7 +34,9 @@
 from cf_units import Unit
 from iris.exceptions import CoordinateNotFoundError
 
-TIME_REFERENCE_DTYPE = np.int64
+from improver.metadata.constants.time_types import (
+    TIME_COORD_NAMES, TIME_REFERENCE_DTYPE, TIME_REFERENCE_UNIT,
+    TIME_DTYPE, TIME_UNIT)
```

Comment:
That's an interesting point.  The distinction I was making when naming those types was between an absolute time (which either iris or cf_units calls `time_reference`, eg via the `is_time_reference` function), which is `seconds since <epoch>`, and a time which is in `seconds`.  But I can see the potential for confusion.

The reason I would prefer not to use `forecast_period` is that it's a specific coordinate name, it might not be the only one we ever apply this to and it might not be one we use forever.  Would maybe `TIME_INTERVAL` be suitable?

Reply:
I've changed this to `TIME_INTERVAL_UNITS` etc, hopefully this is less ambiguous.

Reply:
Yes, `TIME_INTERVAL` and `TIME_INTERVAL_UNITS` are less ambiguous names.